aten/src/ATen/native/LossNLL.cpp

```diff
@@ -23,6 +24,18 @@ inline scalar_t* optional_data(const Tensor& source) {
   return source.defined() ? source.data_ptr<scalar_t>() : nullptr;
 }
 
+
+static int64_t ceil_log2(int64_t x) {
```

Comment:
given that it's used in a lot of places (and will be used in GroupNorm also, #54291), does it make sense to move it to some header and inline? #54291 suggests using native/cpu/utils.h.